dsa-23au/java-dsa/heaps-trees/src/main/java/dev/codewithfriends/BSTMain.java

```diff
@@ -1,13 +1,47 @@
 package dev.codewithfriends;
+import java.util.LinkedList;
+import java.util.List;
+
+public class BSTMain {
+    
+    public static boolean isVisited(BinaryNode node, LinkedList visited) {
+        for (int i=0; i < visited.size(); i++) {
+            if (node.equals(visited.get(i))) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static List<BinaryNode> dfs(BinaryNode node, LinkedList visited) {
```

Comment:
I was struggling quite a bit with grasping this code but after a lot of practice and looking over the textbooks/comments. Binary search trees were very difficult for me because there is a bit you have to keep track of. I wanna use this to represent how crime can be linked to certain areas/counties for my project idea.